custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -629,3 +662,180 @@ def get_queue(self):
+            key="address",
+            vals=[
+                {"name": "address"},
+                {"name": "mac-address"},
+                {"name": "host-name"},
+                {"name": "comment"},
+                {"name": "disabled", "default": True},
+            ],
+            only=[
+                {"key": "disabled", "value": False},
+            ],
+            ensure_vals=[
+                {"name": "address"},
+                {"name": "mac-address"},
+            ]
+        )
+
+        # Also retrieve all entries in ARP table. If some hosts are missing, build it here
+        arp_hosts = parse_api(
+            data={},
+            source=self.api.path("/ip/arp", return_list=True),
+            key="address",
+            vals=[
+                {"name": "address"},
+                {"name": "mac-address"},
+                {"name": "disabled", "default": True},
+                {"name": "invalid", "default": True},
+            ],
+            only=[
+                {"key": "disabled", "value": False},
+                {"key": "invalid", "value": False}
+            ],
+            ensure_vals=[
+                {"name": "address"},
+                {"name": "mac-address"},
+            ]
+        )
+
+        for addr in arp_hosts:
+            if addr not in self.data["accounting"]:
+                self.data["accounting"][addr] = {
+                   "address": arp_hosts[addr]['address'],
+                   "mac-address": arp_hosts[addr]['address']
+                }
+
+        # Build name for host. First try getting DHCP lease comment, then entry in DNS (only static entries)
+        #   and then device's host-name. If everything fails use hosts IP address as name
+        dns_data = parse_api(
+            data={},
+            source=self.api.path("/ip/dns/static", return_list=True),
+            key="address",
+            vals=[
+                {"name": "address"},
+                {"name": "name"},
+            ],
+        )
+
+        for addr in self.data["accounting"]:
+            if str(self.data["accounting"][addr].get('comment', '').strip()):
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['comment']
+            elif addr in dns_data and str(dns_data[addr].get('name', '').strip()):
+                self.data["accounting"][addr]['name'] = dns_data[addr]['name']
+            elif str(self.data["accounting"][addr].get('host-name', '').strip()):
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['host-name']
+            else:
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['address']
+
+        _LOGGER.debug(f"Generated {len(self.data['accounting'])} accounting devices")
+
+        # Build list of local networks
+        dhcp_networks = parse_api(
+            data={},
+            source=self.api.path("/ip/dhcp-server/network", return_list=True),
```

Comment:
return_list=True is now default. Since that used the most. only return_list=False have to be specified